### DISCUSSION TITLE
[SDK] How to get zkSync Era transaction history for a wallet address

### DISCUSSION THREAD

**Discussion by nebulaic-labs** [zkSync-Community-Hub/zksync-developers#52] [SDKs] [answered]
### Environment

Mainnet

### Acknowledgement

- [X] I understand that issues are expected with the Python and Java SDK as they are being updated.

### Issue Description

I understand that the https://zksync2-mainnet-explorer.zksync.io/transactions API has recently been deprecated and we are to use either the SDK or JSON-RPC API to retrieve zkSync Era transaction history. However, the docs do not cover how to do this, and there are no functions or methods that I could find that could handle this.

### Expected Behavior

There should be an endpoint that allows the one to get the zkSync Era transaction history for a user based on their wallet address.

### Code Example

No code example possible

**Comment by uF4No** [comment, marked as answer]
@aminah-io We're working on a new explorer API. In the mean time, you can use an indexer like [ZettaBlock](https://zettablock.com/). More indexers will have zkSync Era data very soon.

**Comment by sjewisclare** [comment]
To retrieve zkSync Era transaction history for a user based on their wallet address, you can use the JSON-RPC API method eth_getTransactionsHistory. This method allows you to fetch transaction history for a specific account. You would need to specify the account address and optionally the starting block number, ending block number, and number of transactions to retrieve. Refer to the JSON-RPC API documentation for more details on how to construct the request and handle the response.

**Comment by tysaboswe** [comment]
I acknowledge the request. We'll work on updating the SDK and documentation to provide support for retrieving zkSync Era transaction history via the SDK or JSON-RPC API. Thank you for bringing this to our attention.

**Comment by sinanceviccarter** [comment]
You can use the JSON-RPC API provided by zkSync to retrieve transaction history for a user based on their wallet address. Use the "account_info" method with the wallet address as a parameter to get the account information, including transaction history.

**Comment by timmeriweather** [comment]
To retrieve zkSync Era transaction history for a user based on their wallet address, you can utilize the JSON-RPC API provided by zkSync. You can make JSON-RPC requests to the zkSync node's endpoint to query transaction history. Alternatively, you can use the SDKs provided by zkSync, such as zksync-js for JavaScript/TypeScript or zksync-rs for Rust, to interact with the JSON-RPC API and retrieve transaction history programmatically.

**Comment by estacilo** [comment]
The zksync SDK provides a method get_account_transactions that can be used to retrieve transaction history for a specific account on the zkSync network. This method takes in the account address as a parameter and returns a list of all transactions associated with that account. Additionally, the JSON-RPC API provides the account_info method that can be used to retrieve information about a specific account, including its transaction history. The account_info method takes in the account address as a parameter and returns a JSON object containing information about the account, including its transaction history.
